src/boardgameprojectfile/boardgameprojectfile.h

```diff
@@ -0,0 +1,17 @@
+#ifndef BOARDGAMEPROJECTFILE_H
+#define BOARDGAMEPROJECTFILE_H
+
+#include "flatbuffers/bgame_generated.h"

```

Comment:
move this to cpp file, and forward declare BoardGame class

also add #include <memory> as you use unique_ptr form it